genesyscloud/supported_content/genesyscloud_supported_content_proxy.go

```diff
@@ -0,0 +1,173 @@
+	return p.createSupportedContentAttr(ctx, p, supportedContent)
+}
+
+// getSupportedContent retrieves all Genesys Cloud supported content
+func (p *supportedContentProxy) getAllSupportedContent(ctx context.Context) (*[]platformclientv2.Supportedcontent, *platformclientv2.APIResponse, error) {
+	return p.getAllSupportedContentAttr(ctx, p)
+}
+
+// getSupportedContentIdByName returns a single Genesys Cloud supported content by a name
+func (p *supportedContentProxy) getSupportedContentIdByName(ctx context.Context, name string) (id string, retryable bool, response *platformclientv2.APIResponse, err error) {
+	return p.getSupportedContentIdByNameAttr(ctx, p, name)
+}
+
+// getSupportedContentById returns a single Genesys Cloud supported content by Id
+func (p *supportedContentProxy) getSupportedContentById(ctx context.Context, id string) (supportedContent *platformclientv2.Supportedcontent, response *platformclientv2.APIResponse, err error) {
+	return p.getSupportedContentByIdAttr(ctx, p, id)
+}
+
+// updateSupportedContent updates a Genesys Cloud supported content
+func (p *supportedContentProxy) updateSupportedContent(ctx context.Context, id string, supportedContent *platformclientv2.Supportedcontent) (*platformclientv2.Supportedcontent, *platformclientv2.APIResponse, error) {
+	return p.updateSupportedContentAttr(ctx, p, id, supportedContent)
+}
+
+// deleteSupportedContent deletes a Genesys Cloud supported content by Id
+func (p *supportedContentProxy) deleteSupportedContent(ctx context.Context, id string) (response *platformclientv2.APIResponse, err error) {
+	return p.deleteSupportedContentAttr(ctx, p, id)
+}
+
+// createSupportedContentFn is an implementation function for creating a Genesys Cloud supported content
+func createSupportedContentFn(ctx context.Context, p *supportedContentProxy, supportedContent *platformclientv2.Supportedcontent) (*platformclientv2.Supportedcontent, *platformclientv2.APIResponse, error) {
+	return p.conversationsApi.PostConversationsMessagingSupportedcontent(*supportedContent)
+}
+
+// getAllSupportedContentFn is the implementation for retrieving all supported content in Genesys Cloud
+func getAllSupportedContentFn(ctx context.Context, p *supportedContentProxy) (*[]platformclientv2.Supportedcontent, *platformclientv2.APIResponse, error) {
+	var allSupportedContents []platformclientv2.Supportedcontent
+	const pageSize = 100
+
+	supportedContents, resp, err := p.conversationsApi.GetConversationsMessagingSupportedcontent(pageSize, 1)
+	if err != nil {
+		return nil, resp, fmt.Errorf("Failed to get supported content: %v", err)
+	}
+	if supportedContents.Entities == nil || len(*supportedContents.Entities) == 0 {
+		return &allSupportedContents, resp, nil
+	}
+	for _, supportedContent := range *supportedContents.Entities {
+		allSupportedContents = append(allSupportedContents, supportedContent)
+	}
```

Comment:
You can replace this for loop with:

```golang
allSupportedContents = append(allSupportedContents, *supportedContents.Entities...)
```

It is not a big deal, but it is something the go linter package highlights. Eventually we'd like for that linter to run against the whole project as part of the workflow and produce zero warnings